File: docs/src/testers/scraping/nested_structures.py
# @sniptest filename=nested_structures.py
# @sniptest show=4-26

from notte_sdk import NotteClient
from pydantic import BaseModel


class Address(BaseModel):
    street: str
    city: str
    country: str


class Company(BaseModel):
    name: str
    description: str
    address: Address
    employee_count: int | None


client = NotteClient()
company = client.scrape(
    "https://example.com/about", response_format=Company, instructions="Extract company information including address"
)

print(Company.address.city)

File: docs/src/testers/scraping/nested_structures.ts
import { z } from 'zod';
import { NotteClient } from 'notte-sdk';

const Address = z.object({
  street: z.string(),
  city: z.string(),
  country: z.string(),
});

const Company = z.object({
  name: z.string(),
  description: z.string(),
  address: Address,
  employee_count: z.number().nullable(),
});

const client = new NotteClient({
  apiKey: process.env.NOTTE_API_KEY,
});

const company = await client.scrape('https://example.com/about', {
  response_format: Company,
  instructions: 'Extract company information including address',
});

console.log(company.address.city);
